### DISCUSSION TITLE
Start search from a base function

### DISCUSSION THREAD

**Discussion by DenisSvirin** [MilesCranmer/PySR#418] [General]
I have a function `F`, that fit the data decently in some cases, but I want to use symbolic regression to find more general solution.
Is there any way to use it as a start formula?

**Comment by MilesCranmer** [comment]
There is, but it’s very low-level. It would basically require you to construct the search state by hand (which is usually used by `warm_start=True`) to include the previously found expression. But maybe we should make it easier to do this automatically?

Another way is to fit the residual between your expression and the ground truth data. But the downside is that you would be enforcing that the learned modification is added to your expression, rather than a full manipulation of it.

Does that make sense?

**Reply by DenisSvirin** [reply]
yes, but the second way doesn't fit my research purpose.

So, basically, I just need to create a csv file with my equation and use `from_file` method?

**Reply by MilesCranmer** [reply]
Unfortunately the csv method doesn’t affect the search state itself. The search state is two Julia objects — a vector of populations, and a hall of fame:

https://github.com/MilesCranmer/SymbolicRegression.jl/blob/4f0b8c576eaccbdcbb6497ab446a3abf6535aaec/src/SymbolicRegression.jl#L1088

So you would need to manually construct the populations (can leave them empty), and the hall of fame (would put the expression here).

This is saved as `raw_julia_state_` inside a PySRRegressor object, and is passed back to the equation search when using `warm_start=True`.

It might be hard for you to set this up yourself. I am unfortunately a bit busy this week as I’m moving countries. But if you ping me the following week I can try to get an example working.

**Reply by u0407** [reply]
Hey Miles, are you free to provide an example to mannualy constuct formulas?

**Reply by MilesCranmer** [reply]
Maybe check out the new template expression spec?

**Reply by u0407** [reply]
ohh I see. Thanks! For anyone who is wondering, here is the link. [https://astroautomata.com/PySR/api/#expression-specifications](url)

**Comment by tomaklutfu** [comment]
I wanted to do a similar thing so I tried two things. First thing was I added the solution from `f` as a column of `X` and it worked somewhat good. Second thing was that devising a custom cost function so that `f` is accentuated more like giving larger weights for close `y` to `f`. This works too but generalizes more so resulting expression has some resemblance to `f` but not exactly.
